### DISCUSSION TITLE
what should I do to lint quickapp ux file

### DISCUSSION THREAD

**Discussion by gengjiawen** [biomejs/biome#3833] [General]
quickapp file (.ux extension) likes vue a lot. What should I do to make biome can lint the file. At least an easy way to lint the script part as js.

```html
<template>
  <div class="tutorial-page">
    <!-- Method 1: Default $item represents the element in the array, $idx represents the index -->
    <div class="tutorial-row" for="{{list}}" tid="uniqueId">
      <text>{{$idx}}.{{$item.name}}</text>
    </div>
    <!-- Method 2: Custom element variable name -->
    <div class="tutorial-row" for="value in list" tid="uniqueId">
      <text>{{$idx}}.{{value.name}}</text>
    </div>
    <!-- Method 3: Custom element and index variable names -->
    <div class="tutorial-row" for="(personIndex, personItem) in list" tid="uniqueId">
      <text>{{personIndex}}.{{personItem.name}}</text>
    </div>
  </div>
</template>

<style lang="less">
  .tutorial-page {
    flex-direction: column;
    .tutorial-row {
      width: 85%;
      margin-top: 10px;
      margin-bottom: 10px;
    }
  }
</style>

<script>
  export default {
    private: {
      list: [
        { name: 'aa', uniqueId: 1},
        { name: 'bb', uniqueId: 2},
        { name: 'cc', uniqueId: 3}
      ]
    },
    onInit () {
      this.$page.setTitleBar({ text: 'Directive for' })
    }
  }
</script>

```

**Comment by dyc3** [comment]
Biome doesn't currently support `.ux` files. Biome *can* format and lint the `script` section of `.vue` files.

I've never heard of quickapp before. Would you mind sharing some more context?

**Reply by gengjiawen** [reply]
> Biome can format and lint the script section of .vue files.

What's minimal work should I do to make biome do this too.

> I've never heard of quickapp before. Would you mind sharing some more context?

Quick Apps are a new type of mobile application format launched by major Chinese smartphone manufacturers like Huawei, Xiaomi, OPPO, and vivo. They aim to provide an app experience that requires no download or installation, similar to mini-programs, but closer to native apps in terms of performance and user experience. Here are some key features of Quick Apps:

1. **Instant Use**: Users can use the app immediately without downloading or installing, significantly reducing barriers and wait times.

2. **Native Experience**: Quick Apps can access the device's native features, offering performance and user experience close to native apps.

3. **Unified Standards**: Jointly launched by multiple manufacturers, they use a unified technical standard, allowing developers to develop once and adapt to multiple devices like phone, watch, foldable .

4. **Lightweight**: Quick Apps are typically much smaller than traditional apps, saving users' storage space.

5. **Easy Development**: Developed using standard front-end technologies (like HTML, CSS, JavaScript), lowering the development threshold.

6. **Convenient Distribution**: Distributed through manufacturers' app stores, users can access them directly through search or other means.

Quick Apps are suitable for scenarios requiring rapid iteration and lightweight services, such as e-commerce event pages, news, and tool applications.


we are currently use eslint. But's it's cumbersome to setup.

**Reply by dyc3** [reply]
Is there any project website, repo, documentation, or specification you could link me to?

**Reply by gengjiawen** [reply]
sure: https://doc.quickapp.cn/. but currently it's lack of english docs.

**Reply by Conaclos** [reply]
If it is similar to Vue, it should not be hard to support it in the same way we support Vue.
One thing I wonder is: is ux widely used in the Chinese community?

**Reply by gengjiawen** [reply]
> One thing I wonder is: is ux widely used in the Chinese community?

Not popular like react and vue. But it definitely deployed to billions of device (includes Huawei, Xiaomi, OPPO, and vivo.) 
It also deployed to vivo watch and xiaomi watch.

**Reply by gengjiawen** [reply]
> If it is similar to Vue, it should not be hard to support it in the same way we support Vue.

What should I do to move this forward ?

**Comment by ematipico** [comment]
For the time being, I am opposed to this proposal. Here's why:

- lack of documentation where the majority of core contributors can read about 
- feature creep: I don't feel comfortable about supporting another syntax (HTML-ish)

I wouldn't underestimate the lack of the documentation, because even though a contributor can decide to send a feature PR, then it's up to the core team to maintain it and fix possible bugs. The lack of community and documentation of this format is a show stopper.

**Reply by gengjiawen** [reply]
I understand lacks english docs will be a problem. Is it possible I can make a plugin to make this work before we have good english docs ?

**Reply by dyc3** [reply]
We haven't implemented plugin support yet, see: #2463
